### PR TITLE
Upgrade CacheInfo Manager

### PR DESCRIPTION
**Pull request by eugeniosant**
Upgrade of Cache Info Manager to facilitate insert of Infinispan connector (into components project).

### CODE REVIEW

File: engine/src/main/java/com/agiletec/aps/system/services/cache/OldCacheManager.java [hunk 1]
```diff
@@ -1,102 +1,98 @@
...
+import com.agiletec.aps.system.common.AbstractService;
+import org.entando.entando.aps.system.services.cache.CacheInfoManager;
+
+/**
+ * Manager of the System Cache
+ *
+ * @author E.Santoboni
+ */
+public class OldCacheManager extends AbstractService implements ICacheManager {
+
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/aps/system/services/cache/OldCacheManager.java]: Kill if possible. Or if not possible or too much work verify mark @Depricated so it doesn't accidentally get used.

File: engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/helper/BaseContentListHelper.java [hunk 2]
```diff
@@ -83,12 +85,13 @@ public String getFilterParam(EntitySearchFilter[] filters) {
...
 	@Cacheable(value = ICacheInfoManager.DEFAULT_CACHE_NAME, 
 			key = "T(com.agiletec.plugins.jacms.aps.system.services.content.helper.BaseContentListHelper).buildCacheKey(#bean, #user)", condition = "#bean.cacheable")
-	@CacheEvict(value = ICacheInfoManager.DEFAULT_CACHE_NAME, 
-			key = "T(com.agiletec.plugins.jacms.aps.system.services.content.helper.BaseContentListHelper).buildCacheKey(#bean, #user)", 
-			beforeInvocation = true, 
-			condition = "T(org.entando.entando.aps.system.services.cache.CacheInfoManager).isExpired(T(com.agiletec.plugins.jacms.aps.system.services.content.helper.BaseContentListHelper).buildCacheKey(#bean, #user))")
+	//@CacheEvict(value = ICacheInfoManager.DEFAULT_CACHE_NAME, 
+	//		key = "T(com.agiletec.plugins.jacms.aps.system.services.content.helper.BaseContentListHelper).buildCacheKey(#bean, #user)", 
+	//		beforeInvocation = true, 
+	//		condition = "T(org.entando.entando.aps.system.services.cache.CacheInfoManager).isExpired(T(com.agiletec.plugins.jacms.aps.system.services.content.helper.BaseContentListHelper).buildCacheKey(#bean, #user))")
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/helper/BaseContentListHelper.java]: Can this be removed?

File: engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java [hunk 3]
```diff
@@ -100,13 +103,14 @@ public String getShowletParam(EntitySearchFilter[] filters) {
...
             condition = "#bean.cacheable && !T(com.agiletec.plugins.jacms.aps.system.services.content.widget.ContentListHelper).isUserFilterExecuted(#bean)")
-    @CacheEvict(value = ICacheInfoManager.DEFAULT_CACHE_NAME,
-            key = "T(com.agiletec.plugins.jacms.aps.system.services.content.widget.ContentListHelper).buildCacheKey(#bean, #reqCtx)",
-            beforeInvocation = true,
-            condition = "T(org.entando.entando.aps.system.services.cache.CacheInfoManager).isExpired(T(com.agiletec.plugins.jacms.aps.system.services.content.widget.ContentListHelper).buildCacheKey(#bean, #reqCtx))")
+    //@CacheEvict(value = ICacheInfoManager.DEFAULT_CACHE_NAME,
+    //        key = "T(com.agiletec.plugins.jacms.aps.system.services.content.widget.ContentListHelper).buildCacheKey(#bean, #reqCtx)",
+    //        beforeInvocation = true,
+    //        condition = "T(org.entando.entando.aps.system.services.cache.CacheInfoManager).isExpired(T(com.agiletec.plugins.jacms.aps.system.services.content.widget.ContentListHelper).buildCacheKey(#bean, #reqCtx))")
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java]: Can this be removed?

File: engine/src/test/java/org/entando/entando/aps/system/services/cache/CacheInfoManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+			}
+			@Override
+			public Class<? extends Annotation> annotationType() {
+				return CacheInfoEvict.class;
+			}
+		};
+        cacheInfoManager.aroundCacheInfoEvictMethod(proceedingJoinPoint, cacheInfoEvict);
+        Mockito.verify(proceedingJoinPoint, Mockito.times(1)).getTarget();
+        //Mockito.verify(cacheInfoManager, Mockito.times(2))
+		//		.flushGroup(Mockito.any(String.class), Mockito.any(String.class));
```
Comment (joewhite101) [inline engine/src/test/java/org/entando/entando/aps/system/services/cache/CacheInfoManagerTest.java]: Remove if not needed?

File: engine/src/test/java/org/entando/entando/aps/system/services/cache/CacheInfoManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+
+/**
+ * @author E.Santoboni
+ */
+public class CacheInfoManagerTest {
```
Comment (joewhite101) [inline engine/src/test/java/org/entando/entando/aps/system/services/cache/CacheInfoManagerTest.java]: These tests look really good. Probably worth thinking about how we can generalize these tests to apply to all of the cache managers generically so we don't have to write the same test methods 50+ times.

Doesn't need to happen now but keep in mind.

**Comment by entando-service** [comment]
All is well

File: engine/src/test/java/org/entando/entando/aps/system/services/cache/CacheInfoManagerTest.java (hunk 5 above)
Comment (eugeniosant) [reply engine/src/test/java/org/entando/entando/aps/system/services/cache/CacheInfoManagerTest.java]: Clear concept.
This is the only case in which I thought to not insert a CacheWrapper, for the simple reason that this manager already manages cache.
We will keep this in mind for the other issues.

File: engine/src/test/java/org/entando/entando/aps/system/services/cache/CacheInfoManagerTest.java (hunk 4 above)
Comment (eugeniosant) [reply engine/src/test/java/org/entando/entando/aps/system/services/cache/CacheInfoManagerTest.java]: I forget it, I'll fix

File: engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java (hunk 3 above)
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java]: I'll remove it

File: engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/helper/BaseContentListHelper.java (hunk 2 above)
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/helper/BaseContentListHelper.java]: Yes, I'll fix

File: engine/src/main/java/com/agiletec/aps/system/services/cache/OldCacheManager.java (hunk 1 above)
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/aps/system/services/cache/OldCacheManager.java]: Not to much work, I'll remove it

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Pull request merged**